go/vt/vtgate/endtoend/lookup_test.go

```diff
@@ -260,6 +260,379 @@ func TestHashLookupMultiInsertIgnore(t *testing.T) {
 	}
 }
 
+func TestSecondaryLookup(t *testing.T) {
+	ctx := context.Background()
+	conn, err := mysql.Connect(ctx, &vtParams)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+
+	// connShard1 is for queries that target shards.
+	connShard1, err := mysql.Connect(ctx, &vtParams)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer connShard1.Close()
+
+	// connShard2 is for queries that target shards.
+	connShard2, err := mysql.Connect(ctx, &vtParams)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer connShard2.Close()
+
+	// insert multiple values
+	exec(t, conn, "begin")
+	exec(t, conn, "insert into t3(user_id, lastname, address) values(1,'snow','castle_black'), (2,'stark','winterfell')")
+	exec(t, conn, "commit")
+	qr := exec(t, conn, "select * from t3")
+	if got, want := fmt.Sprintf("%v", qr.Rows), "[[INT64(1) VARCHAR(\"snow\") VARCHAR(\"castle_black\")] [INT64(2) VARCHAR(\"stark\") VARCHAR(\"winterfell\")]]"; got != want {
```

Comment:
Is the order of rows expected to be stable?

Reply:
Added order by clause